frontend/src/pages/RunDetailsRouter.tsx

```diff
@@ -16,45 +16,80 @@
 
 import React from 'react';
 import { useQuery } from 'react-query';
-import { ApiRunDetail } from 'src/apis/run';
+import { ApiJob } from 'src/apis/job';
+import { ApiResourceType, ApiRunDetail } from 'src/apis/run';
 import { RouteParams } from 'src/components/Router';
 import { Apis } from 'src/lib/Apis';
 import * as WorkflowUtils from 'src/lib/v2/WorkflowUtils';
 import EnhancedRunDetails, { RunDetailsProps } from 'src/pages/RunDetails';
 import { RunDetailsV2 } from 'src/pages/RunDetailsV2';
 
+// Retrieve the Job ID from run if it is a run of recurring run
+function getJobID(apiRun: ApiRunDetail | undefined): string | undefined {
```

Comment:
NIT: There are some legacy codes and APIs which refers to a `Job`, let's try to replace them with `RecurringRun` as much as we can. We have abandoned the terminology of `Job` and we are in progress to move away from it.